CONTRIBUTING.md

```diff
@@ -69,8 +69,6 @@ Run all static analysis tools with:
 make lint
 ```
 
-This may make changes to the local files if improvements are available.
```

Comment:
Oh hang on -- Does this mean the linter will not change any code locally?

Perhaps we should introduce pre-commit to do that instead?

Reply:
or keep pre-commit optional and have a separate format-ci command

Reply:
We need to decide if `make lint` checks formatting or applies formatting - it can't do both. We need the former in CI so I recommend that.

For local formatting, I would recommend both pre-commit and advising people to set-up ruff in their editors.

Reply:
> it can't do both

It can. using pre-commit would allow us to have it make changes _and_ return non-0 if changes were made.

This has the benefit of being able to then show the changes in the CI output if we want

Reply:
Sure, if you can solve the problem by running pre-commit then that's great. 

Will merge this for now to solve the immediate problem.